frontend/src/metabase/static-viz/components/LineAreaBarChart/utils/series.unit.spec.ts

```diff
@@ -0,0 +1,401 @@
+import { merge } from "icepick";
+import { colors } from "metabase/lib/colors";
+import type {
+  ChartSettings,
+  SeriesWithOneOrLessDimensions,
+  SeriesWithTwoDimensions,
+} from "../../XYChart/types";
+import { getSeriesWithColors } from "./series";
+
+const settings: ChartSettings = {
+  x: {
+    type: "ordinal",
+  },
+  y: {
+    type: "linear",
+  },
+  labels: {
+    left: "Count",
+    bottom: "Date",
+  },
+};
+
+describe("getSeriesWithColors", () => {
+  it("should return an empty series given an empty series", () => {
+    const seriesWithColors = getSeriesWithColors([], settings, getPalette({}));
+
+    expect(seriesWithColors).toEqual([]);
+  });
+
+  describe("Series without ones or less dimensions", () => {
+    const multipleSeries: SeriesWithOneOrLessDimensions[][] = [
```

Comment:
Sampled from this chart
![image](https://user-images.githubusercontent.com/1937582/201139336-03941133-b15e-4927-bdc0-51aebc6f0e39.png)